raiden/tests/unit/network/rtc/test_web_rtc.py

```diff
@@ -66,3 +68,49 @@ def _send_to_2(*args, **kwargs) -> None:  # type: ignore
 
     while manager1.has_ready_channel(address2):
         gevent.sleep(0.1)
+
+    manager1.stop()
+    manager2.stop()
+
+
+def _get_file_descriptors() -> List:
+    pid = os.getpid()
+    return os.listdir(f"/proc/{pid}/fd/")
+
+
+@pytest.mark.skipif(sys.platform != "linux", reason="this is a Linux-only test")
+@pytest.mark.skipif(sys.version_info < (3, 8), reason="it seems we leak on Python < 3.8")
+def test_leak_file_descriptors(monkeypatch: Any) -> None:
+    address1 = make_signer().address
+    address2 = make_signer().address
+    stop_event1 = Event()
+    stop_event2 = Event()
+
+    def _noop_send(*args, **kwargs) -> None:  # type: ignore
+        pass
+
+    manager1 = WebRTCManager(address1, _ignore_web_rtc_messages, _noop_send, stop_event1)
+    manager2 = WebRTCManager(address2, _ignore_web_rtc_messages, _noop_send, stop_event2)
+
+    assert not manager1.has_ready_channel(address2), "no channel should exist yet"
+    assert not manager2.has_ready_channel(address1), "no channel should exist yet"
+
+    # reduce the channel init timeout so that the test does not take too long
+    monkeypatch.setattr(manager1, "get_channel_init_timeout", lambda *_: 3.0)
+
+    max_open_fds = 0
+    for x in range(20):
```

Comment:
Can we get a name better than `x`? What are we counting?

Reply:
Just iterations.